### PR TITLE
Update serial_driver.md

### PR DESCRIPTION
**Pull request by dunk2k**
## Description

- wording and terms changed to prevent confusion or ambiguity,  and assist language translators
- Clarify conditions of when to use half-duplex or full-duplex
- numbering fixed where broken
- correct steps flow in *-duplex sections
- confirm sooner rather than later omission of pull-up resistor for Half-duplex with RP2040

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/serial_driver.md [hunk 1]
```diff
@@ -74,25 +74,25 @@ Targeting ARM boards based on ChibiOS, where communication is offloaded to a USA
...
+To use the Half-duplex driver follow these steps to activate it. If you target the Raspberry Pi RP2040 PIO implementation, start at step 1a.
 
 1. Change the `SERIAL_DRIVER` to `usart` in your keyboards `rules.mk` file:
 
 ```make
 SERIAL_DRIVER = usart
 ```
 
-2. (RP2040 PIO only!) Change the `SERIAL_DRIVER` to `vendor` in your keyboards `rules.mk` file:
+1a. (RP2040 + PIO only!) Change the `SERIAL_DRIVER` to `vendor` in your keyboards `rules.mk` file:
```
Comment (fauxpark) [inline docs/serial_driver.md:89]: This breaks the markdown formatting.

File: docs/serial_driver.md [hunk 2]
```diff
@@ -129,27 +129,27 @@ Targeting ARM boards based on ChibiOS where communication is offloaded to an USA
...
+To use the Full-duplex driver follow these steps to activate it. If you target the Raspberry Pi RP2040 PIO implementation, start at step 1a.
 
 1. Change the `SERIAL_DRIVER` to `usart` in your keyboards `rules.mk` file:
 
 ```make
 SERIAL_DRIVER = usart
 ```
 
-2. (RP2040 PIO only!) Change the `SERIAL_DRIVER` to `vendor` in your keyboards `rules.mk` file:
+1a. (RP2040 + PIO only!) Change the `SERIAL_DRIVER` to `vendor` in your keyboards `rules.mk` file:
```
Comment (fauxpark) [inline docs/serial_driver.md:146]: Same here.

File: docs/serial_driver.md (hunk 1 above)
Comment (dunk2k) [reply docs/serial_driver.md]: thanks for spotting - fixed as of commit [794caed](https://github.com/qmk/qmk_firmware/pull/23035/commits/794caed80691d7eb14575f0f205294e4edfbaa92)

**Review by KarlK90 — APPROVED** [review]
@dunk2k Thank you for the overhaul! As a non native English speaker it is too easy to miss these subtleties.

File: docs/serial_driver.md [hunk 3]
```diff
@@ -225,17 +229,17 @@ Just below `#include_next <mcuconf.h>` add:
...
+The `PIO` subsystem is a Raspberry Pi RP2040 specific implementation, using an integrated PIO peripheral and is therefore only available on this MCU. Because of the flexible nature of PIO peripherals, **any** GPIO pin can be used as a `TX` or `RX` pin. Half-duplex and Full-duplex operation modes are fully supported with this driver. Half-duplex uses the built-in pull-ups and GPIO manipulation of the RP2040 to drive the line high by default, thus an external pull-up resistor **is not required**.
 
-You may optionally switch the PIO peripheral used with the following define in config.h:
+Optionally, the PIO peripheral to be utilised for split communication can be changed with the following define in config.h:
```
Comment (drashna) [inline docs/serial_driver.md:242]:
```suggestion
Optionally, the PIO peripheral utilized for split communication can be changed with the following define in config.h:
```

**Review by drashna — APPROVED** [review]

**Review by KarlK90 — APPROVED** [review]

**Pull request merged**